Improve Schedules Build form labels and layouts

toolmantim opened the pull request:
Some cleanups before we announce scheduled builds are live:

* Clarifies which fields are required
* Puts the description at the top before all the Hard Stuff
* Added more information about the the cron interval syntax, along with `@` extensions we support
* Made sure teams forms were consistent

There's some things that still need improving which people might ask:

* Such as "How do I change/update the "Creator" of the builds?"
* What happens if the "Creator" is removed from the org?

Before:

<img width="877" alt="before-1" src="https://cloud.githubusercontent.com/assets/153/26183243/f51396e4-3bc0-11e7-8c92-e0c8a48b689c.png">

After:

<img width="874" alt="after-1" src="https://cloud.githubusercontent.com/assets/153/26183248/f9ae3308-3bc0-11e7-9a72-98c4f21dabfc.png">

Before:

<img width="883" alt="before-2" src="https://cloud.githubusercontent.com/assets/153/26183260/08a3e290-3bc1-11e7-82c4-983d47514b6b.png">

After:

<img width="891" alt="after-2" src="https://cloud.githubusercontent.com/assets/153/26183262/0b1cb18c-3bc1-11e7-94dd-e8ca8431fd97.png">

Before:

<img width="916" alt="before-3" src="https://cloud.githubusercontent.com/assets/153/26183264/0fc1dd3e-3bc1-11e7-9afc-8b12d0185f64.png">

After:

<img width="933" alt="after-3" src="https://cloud.githubusercontent.com/assets/153/26183266/1385d308-3bc1-11e7-9834-56610fdba4b8.png">

harrietgrace commented on app/components/pipeline/schedules/Form.js:
```diff
@@ -30,50 +30,50 @@ class Form extends React.Component {
-        <FormTextField
           label="Message"
-          help="The message to use for the created build"
+          help="The message to use for the build. Default: “Scheduled Build for (Pipeline Name)”"
```
Does that default auto-populate? Because it wasn't appearing in the screenshots of the hourly build?

harrietgrace commented on app/components/pipeline/schedules/Form.js:
```diff
@@ -30,50 +30,50 @@ class Form extends React.Component {
         <FormTextField
           label="Branch"
-          help="The branch to use for the created build"
+          help={`The branch to use for the build. Default: the pipeline’s default branch (currently “${this.props.pipeline.defaultBranch}”)`}
```
Niiiice 👌🏻 What do you think about saying 'yours is currently __'? Otherwise people might think this help text is generic?

toolmantim commented:
That's awesome feedback @harrietgrace! I've just updated it again, and moved it back to using placeholders:

<img width="878" alt="after-5" src="https://cloud.githubusercontent.com/assets/153/26185973/ab27648c-3bd1-11e7-93f4-274df5096325.png">

What do you reckon?

ticky commented:
“Placeholders as defaults” feels like a strong UI move we’ve made, so good to keep it consistent!

toolmantim commented:
@ticky oh yeah. Strong in this PR, or that's a general pattern we already use? We ended up deciding just to prefill the fields instead.

ticky commented:
It’s a pattern we already use! The new builds popup and such all show their fallback when empty! It’s a good use of the placeholder, IMO!
